hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/placement/CSMappingPlacementRule.java

```diff
@@ -228,7 +228,11 @@ private VariableContext createVariableContext(
       ApplicationSubmissionContext asc, String user) {
     VariableContext vctx = new VariableContext();
 
-    vctx.put("%user", cleanName(user));
+    String cleanedName = cleanName(user);
+    if (!user.equals(cleanedName)) {
+      vctx.putOriginal("%user", user);
+    }
+    vctx.put("%user", cleanedName);
```

Comment:
Nit: is this actually needed? cleanName returns the original name if it contains no dots, and it seems like the same logic is implemented here. Previous uses of cleanName also didn't create this logic separately.

Reply:
Yeah, the whole point is that if we don't keep the originalName separately (e.g. test.user) then the VariableMatcher will try to match for test_dot_user instead) so subsequent mapping rule actions won't be applied.

Just try this out by commenting out the code block you think is unnecessary, the test called TestCSMappingPlacementRule#testOriginalUserNameWithDotCanBeUsedInMatchExpression will fail immediately)

Reply:
Ahh I've missed the putOriginal/put method differences, feel free to disregard this one.

Reply:
sure :)